source/common/config/grpc_mux_impl.cc

```diff
@@ -206,6 +206,9 @@ void GrpcMuxImpl::onReceiveMessage(std::unique_ptr<envoy::api::v2::DiscoveryResp
       resources.emplace(resource_name, resource);
     }
     for (auto watch : api_state_[type_url].watches_) {
+      // onConfigUpdate should be called in all cases for single watch xDS (Cluster and Listener)
+      // even if the message does not have resources so that update_empty stat is properly
+      // incremented.
       if (watch->resources_.empty()) {
```

Comment:
@htuch single watch xDS case is already handled here. So just added a comment to be clear. Also added a test which validates that.